src/oscar/apps/catalogue/receivers.py

```diff
@@ -3,34 +3,28 @@
 from django.conf import settings
 
 if settings.OSCAR_DELETE_IMAGE_FILES:
-
-    from oscar.core.loading import get_model
-
     from django.db import models
-    from django.db.models.signals import post_delete
+    from django.db.models.signals import pre_delete
 
-    from sorl import thumbnail
-    from sorl.thumbnail.helpers import ThumbnailError
+    from oscar.core.loading import get_model
+    from oscar.core.thumbnails import get_thumbnailer
 
     ProductImage = get_model('catalogue', 'ProductImage')
     Category = get_model('catalogue', 'Category')
 
     def delete_image_files(sender, instance, **kwargs):
         """
-        Deletes the original image, created thumbnails, and any entries
-        in sorl's key-value store.
+        Deletes the original image and created thumbnails.
         """
-        image_fields = (models.ImageField, thumbnail.ImageField)
+        image_fields = (models.ImageField,)
+        thumbnailer = get_thumbnailer()
         for field in instance._meta.fields:
             if isinstance(field, image_fields):
                 # Make Django return ImageFieldFile instead of ImageField
-                fieldfile = getattr(instance, field.name)
-                try:
-                    thumbnail.delete(fieldfile)
-                except ThumbnailError:
-                    pass
+                field_file = getattr(instance, field.name)
+                thumbnailer.delete_thumbnails(field_file)
 
-    # connect for all models with ImageFields - add as needed
+    # Connect for all models with ImageFields - add as needed
     models_with_images = [ProductImage, Category]
     for sender in models_with_images:
-        post_delete.connect(delete_image_files, sender=sender)
+        pre_delete.connect(delete_image_files, sender=sender)
```

Comment:
What is the reason for changing this to `pre_delete`?

Reply:
Returned to `post_delete` with adding `save=False` to `delete` method of thumbnailer (`easy-thumbnails`).